rebar.config.erl

```diff
@@ -268,3 +268,30 @@ str(B) when is_binary(B) -> unicode:characters_to_list(B, utf8).
 
 erl_opts_i() ->
     [{i, "apps"}] ++ [{i, Dir}  || Dir <- filelib:wildcard(filename:join(["apps", "**", "include"]))].
+
+dialyzer(Config) ->
+    {dialyzer, OldDialyzerConfig} = lists:keyfind(dialyzer, 1, Config),
+
+    AppsToAnalyse = case os:getenv("DIALYZER_ANALYSE_APP") of
+        false ->
+            [];
+        Value ->
+            [ list_to_atom(App) || App <- string:tokens(Value, ",")]
+    end,
+
+    AppsDir = filename:join(["apps"]),
+    AppNames = list_dir(AppsDir),
+    AppsToExclude = AppNames -- AppsToAnalyse,
+
+    case length(AppsToAnalyse) > 0 of
+        true ->
+            lists:keystore(dialyzer, 1, Config, {dialyzer, OldDialyzerConfig ++ [{exclude_apps, AppsToExclude}]});
+        false ->
+            lists:keystore(dialyzer, 1, Config, {dialyzer, OldDialyzerConfig})
```

Comment:
no need to put the old config back, variables in Erlang are immutable.
i.e. we can simply return `Config` here.